### PR TITLE
Update miri submodule

### LINKED ISSUE #101344: `miri` no longer builds after rust-lang/rust#101333
Hello, this is your friendly neighborhood mergebot.
After merging PR rust-lang/rust#101333, I observed that the tool miri has failing tests.
A follow-up PR to the repository https://github.com/rust-lang/miri is needed to fix the fallout.

cc @matthiaskrgr, do you think you would have time to do the follow-up work?
If so, that would be great!

### PR DESCRIPTION
**Pull request by oli-obk**
fixes #101344

cc @rust-lang/miri 

r? @ghost

### CODE REVIEW

File: compiler/rustc_middle/src/mir/mod.rs [hunk 1]
```diff
@@ -1478,7 +1478,9 @@ impl<'tcx> Place<'tcx> {
     /// It's guaranteed to be in the first place
     pub fn has_deref(&self) -> bool {
         // To make sure this is not accidently used in wrong mir phase
-        debug_assert!(!self.projection[1..].contains(&PlaceElem::Deref));
+        debug_assert!(
+            self.projection.is_empty() || !self.projection[1..].contains(&PlaceElem::Deref)
+        );
```
Comment (RalfJung) [inline compiler/rustc_middle/src/mir/mod.rs:1483]: Oh, so *that* was the actual cause of the strange ICE...
Cc @ouz-a

**Comment by RalfJung** [comment]
@bors p=4
Miri has been missing for more than a week.

**Comment by RalfJung** [comment]
Ah dang, https://github.com/rust-lang/rust/pull/101476 will probably break Miri again...

File: compiler/rustc_middle/src/mir/mod.rs (hunk 1 above)
Comment (ouz-a) [reply compiler/rustc_middle/src/mir/mod.rs:1483]: I wonder how this slipped without causing any issues until now.

Comment (RalfJung) [reply compiler/rustc_middle/src/mir/mod.rs:1483]: Looks like the ui test suite never hits the code path where this is empty.

**Comment by oli-obk** [comment]
ok this is just breakage from my other PR... doing a miri rustup, then refreshing this PR

**Pull request merged**